src/util/floatingpoint.h

```diff
@@ -115,30 +117,164 @@ namespace CVC4 {
   }; /* struct RoundingModeHashFunction */
 
 
+
   /**
-   * A concrete floating point number
+   * Use symfpu to handle literals
    */
+  namespace symfpuLiteral {
+   // Forward declaration
+    template <bool T> class wrappedBitVector;
+    
+    // Wrap up the types into one template parameter
+    class traits {
+    public :
+      typedef unsigned bwt;
+      typedef bool prop;
+      typedef ::CVC4::RoundingMode rm;
+      typedef ::CVC4::FloatingPointSize fpt;
+      typedef wrappedBitVector< true> sbv;
+      typedef wrappedBitVector<false> ubv;
+
+      static rm RNE (void);
+      static rm RNA (void);
+      static rm RTP (void);
+      static rm RTN (void);
+      static rm RTZ (void);
+
+      static void precondition(const prop &p);
+      static void postcondition(const prop &p);
+      static void invariant(const prop &p);
+    };
+
+    typedef traits::bwt bwt;
+    typedef traits::prop prop;
+    typedef traits::rm rm;
+    typedef traits::fpt fpt;
+    typedef traits::ubv ubv;
+    typedef traits::sbv sbv;
+
+      // Type function
+    template <bool T> struct signedToLiteralType;
+
+    template <> struct signedToLiteralType< true> {
+      typedef int literalType;
+    };
+    template <> struct signedToLiteralType<false> {
+      typedef  unsigned int literalType;
+    };
+
+
+    template <bool isSigned>
+    class wrappedBitVector : public BitVector {
```

Comment:
Reorder according to the style guidelines:
https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#names-and-order-of-includes

Reply:
Headers re-ordered.  (I'm not sure the connection to this line though...)